lib/open_project/text_formatting.rb

```diff
@@ -418,5 +425,19 @@ def replace_toc(text, headings)
         end
       end
     end
+
+    #
+    # displays the current url plus an optional anchor
+    #
+    def full_url(anchor_name = '')
+      return false if current_request.nil?
```

Comment:
You don't need to do that. I am fine with `url.nil?`

My point is that mixed return types ring my statically typed alarm bells. Even more if they are not documented.